src/ipc/ipc4/ams_helpers.c

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Ievgen Ganakov <ievgen.ganakov@intel.com>
+ */
+
+#include <ipc4/ams_helpers.h>
+#include <sof/audio/component.h>
+
+#if CONFIG_AMS
+
+int ams_helper_register_producer(const struct comp_dev *dev,
+				 uint32_t ams_uuid_id,
+				 const uint8_t *msg_uuid)
+{
+	uint16_t mod_id = IPC4_MOD_ID(dev_comp_id(dev));
+	uint16_t inst_id = IPC4_INST_ID(dev_comp_id(dev));
+	int ret;
+
+	ret = ams_get_message_type_id(msg_uuid, &ams_uuid_id);
+	if (ret)
+		return ret;
+
+	return ams_register_producer(ams_uuid_id, mod_id, inst_id);
+}
+
+int ams_helper_unregister_producer(const struct comp_dev *dev,
+				   uint32_t ams_uuid_id)
+{
+	uint16_t mod_id = IPC4_MOD_ID(dev_comp_id(dev));
+	uint16_t inst_id = IPC4_INST_ID(dev_comp_id(dev));
+
+	return ams_unregister_producer(ams_uuid_id, mod_id, inst_id);
+}
+
+int ams_helper_register_consumer(struct comp_dev *dev,
+				 uint32_t ams_uuid_id,
+				 const uint8_t *msg_uuid,
+				 ams_msg_callback_fn callback)
+{
+	uint16_t mod_id = IPC4_MOD_ID(dev_comp_id(dev));
+	uint16_t inst_id = IPC4_INST_ID(dev_comp_id(dev));
+	int ret;
+
+	ret = ams_get_message_type_id(msg_uuid, &ams_uuid_id);
```

Comment:
I think you want the caller of this function to get the UUID ID back? Then the second parameter of this function should be a pointer. Same in `ams_helper_register_producer()`. Has this been runtime-tested?..

Reply:
@lyakh Yes, it has been tested locally and ams is working. But I hit another issue related to IPC4 notification in KD which is under investigation now. Definitely it is a mistake, and it's already fixed. Thanks!